projectional/src/main/java/jetbrains/jetpad/projectional/selection/Selection.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2012-2016 JetBrains s.r.o
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package jetbrains.jetpad.projectional.selection;
+
+import jetbrains.jetpad.cell.Cell;
+
+public class Selection {
+  public final Cell start;
+  public final Object startPosition;
+  public final Cell end;
+  public final Object endPosition;
+
+  public Selection(Cell start, Object startPosition, Cell end, Object endPosition) {
```

Comment:
It is intended to be cell-specific position object. Like integer for text
cell, Xpath for a DOM element etc. I expect 1-1 relationship between cell
class and position type. It is possible to make it generic, but
SelectionListener would expose Seelection<?,?> anyway.
19 Апр 2016 г. 20:35 пользователь "Timur Zambalayev" <
notifications@github.com> написал:

> In
> projectional/src/main/java/jetbrains/jetpad/projectional/selection/Selection.java
> <https://github.com/JetBrains/jetpad-projectional/pull/237#discussion_r60275176>
> :
>
> > + * distributed under the License is distributed on an "AS IS" BASIS,
> > + * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > + * See the License for the specific language governing permissions and
> > + * limitations under the License.
> > + */
> > +package jetbrains.jetpad.projectional.selection;
> > +
> > +import jetbrains.jetpad.cell.Cell;
> > +
> > +public class Selection {
> > +  public final Cell start;
> > +  public final Object startPosition;
> > +  public final Cell end;
> > +  public final Object endPosition;
> > +
> > +  public Selection(Cell start, Object startPosition, Cell end, Object endPosition) {
>
> What will we have in startPosition and endPosition? Objects of what class?
> Why is it Object?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/JetBrains/jetpad-projectional/pull/237/files/2628f5e56358c5e0e6538e95dd47cb1edce4d388#r60275176>
>